### PR TITLE
append label and remove the label for the same function

### LINKED ISSUE #351: Labels added on operandRequest and operandBindInfo should be appended instead of overwrite
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- ODLM version:
- Minikube/KIND/OCP version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by horis233**
**What this PR does / why we need it**:

This PR is used to append label when the OperandRequest an OperandBindInfo have existing labels

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #351 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by chenzhiwei** [comment]
why removing the exist labels?

**Comment by horis233** [comment]
@chenzhiwei  It is for this issue https://github.com/IBM/operand-deployment-lifecycle-manager/issues/329. If the RegistryNamespace or RegistryName changed in OperandRequest or OperandBindInfo CR, the old label won't be removed.

File: pkg/apis/operator/v1alpha1/operandbindinfo_types.go [hunk 1]
```diff
@@ -129,6 +130,16 @@ func (r *OperandBindInfo) SetDefaultsRequestSpec() {
...
+		r.Labels = make(map[string]string)
+	} else {
+		reg, _ := regexp.Compile(`^(.*)\.(.*)$\/registry`)
+		for label := range r.Labels {
+			if reg.MatchString(label) {
+				delete(r.Labels, label)
+			}
+		}
+	}
 	r.Labels = make(map[string]string)
```
Comment (DanielXLee) [inline pkg/apis/operator/v1alpha1/operandbindinfo_types.go:143]: this line should be delete

Comment (horis233) [reply pkg/apis/operator/v1alpha1/operandbindinfo_types.go]: @DanielXLee Thanks, nice catch :)

**Pull request merged**